In [1]:
#!python -m pip install -U python-woc pandas matplotlib
# please clear the output of this cell in your notebook before checking it in

In [2]:
from woc.remote import WocMapsRemote
from tqdm import tqdm
import pandas as pd
# creates the client
woc = WocMapsRemote( base_url="https://worldofcode.org/api/")
# if you got an API key
# woc = WocMapsRemote( base_url="https://worldofcode.org/api/", api_key="woc-XXXXXX-YYYYYY" )

# Now for each of the ten projects you were assigned get commits, e.g.
projects = ['WorldWideTelescope/pywwt', 'chakravala/Grassmann.jl', 'roboticslibrary/rl', 'ROBelgium/MSNoise', 'stepjam/RLBench', 'FragIt/fragit-main', 'architecture-building-systems/revitpythonshell', 'VLSIDA/OpenRAM', 'kaczmarj/neurodocker', 'code-iai/ROSIntegration']
list_df_commits = []
for prj in projects:
  # Convert GitHub repository names to WoC V2412 project names
  prj = prj.lower().replace('/','_',2)
  commits = woc.get_values('p2c', prj)
  df = pd.DataFrame(commits, columns=["sha1"])
  df['project'] = prj
  list_df_commits.append(df)
df = pd.concat(list_df_commits)
#perhaps save the list (if no errors) so you do not need to retrieve them again
df.to_csv('df_commits.csv')
df.head(1)

,sha1,project
0,000287679e1bef118cffcc087810c3fea90b6dfe,worldwidetelescope_pywwt


In [3]:
# If the number of commits its not very large, you can try to get them all at the same time,
# The max batch size is 10
import time
# let us first split df['sha1'] in chunks
chunks = [df['sha1'][x:x+10] for x in range(0, len(df), 10)]
commit_data = []

for chunk in tqdm(chunks): # iterate over the commits
  # res, err = woc.show_content_many('commit',chunk.to_list())

  # commit.tch returns the same data but faster
  res, err = woc.get_values_many('commit.tch',chunk.to_list())
  res = {k: v[0] for k, v in res.items()}  # this conversion is necessary because of the internal implementation

  # to walk around the rate limit
  time.sleep(1)

  if err: # check for errors
    print('Got Errors', err)

  for commit_sha, commit in res.items():
    # flatten commit objects
    commit_data.append({
          'commit': commit_sha,
          'tree': commit[0],
          'parent': list(commit[1]),
          'author': commit[2][0],
          'author_time': int(commit[2][1]),
          'author_tz': commit[2][2],
          'committer': commit[3][0],
          'committer_time': int(commit[3][1]),
          'committer_tz': commit[3][2],
          'message': commit[4],
    })

df_commit_data = pd.DataFrame(commit_data)
df_commit_data = df_commit_data.merge(df, left_on='commit', right_on='sha1')
df_commit_data.to_csv('df_commit_data.csv', index=False)
df_commit_data.head(2)

  0%|▏                                                    | 4/1477 [00:04<25:15,  1.03s/it]

Got Errors {'058b65fbfffa0af1168fd0fe09465a82e3488d69': 'Key 058b65fbfffa0af1168fd0fe09465a82e3488d69 not found in /da5_fast/All.sha1c/commit_5.tch'}


  1%|▍                                                   | 11/1477 [00:11<24:56,  1.02s/it]

Got Errors {'0ea134e94428a4d551221e76c7f64b761b52af5e': 'Key 0ea134e94428a4d551221e76c7f64b761b52af5e not found in /da5_fast/All.sha1c/commit_14.tch'}


  1%|▋                                                   | 18/1477 [00:18<24:47,  1.02s/it]

Got Errors {'17acdd2be12fdb7838c14083303307b5fd1a21dd': 'Key 17acdd2be12fdb7838c14083303307b5fd1a21dd not found in /da5_fast/All.sha1c/commit_23.tch'}


  2%|▉                                                   | 25/1477 [00:25<24:40,  1.02s/it]

Got Errors {'2266224c74e0b155b6d0e890872b0ae514846431': 'Key 2266224c74e0b155b6d0e890872b0ae514846431 not found in /da5_fast/All.sha1c/commit_34.tch'}


  3%|█▌                                                  | 45/1477 [00:45<24:16,  1.02s/it]

Got Errors {'434c9687608d402f76c6c8852ead7df1ff8b4624': 'Key 434c9687608d402f76c6c8852ead7df1ff8b4624 not found in /da5_fast/All.sha1c/commit_67.tch'}


  3%|█▋                                                  | 49/1477 [00:49<24:21,  1.02s/it]

Got Errors {'48621b10a88e9e3390639bde39a647796a351b4c': 'Key 48621b10a88e9e3390639bde39a647796a351b4c not found in /da5_fast/All.sha1c/commit_72.tch'}


  4%|██▏                                                 | 62/1477 [01:03<24:03,  1.02s/it]

Got Errors {'5b6d94f7e7851ecf3d6f1017a21377802ae6fb8d': 'Key 5b6d94f7e7851ecf3d6f1017a21377802ae6fb8d not found in /da5_fast/All.sha1c/commit_91.tch'}


  5%|██▋                                                 | 78/1477 [01:19<23:45,  1.02s/it]

Got Errors {'6fceee91debc826acacbd1e4c6391983e3c92504': 'Key 6fceee91debc826acacbd1e4c6391983e3c92504 not found in /da5_fast/All.sha1c/commit_111.tch'}


  6%|███                                                 | 87/1477 [01:28<23:37,  1.02s/it]

Got Errors {'7df15425872b27bcf6e4b5bd6f2daee80cd1b42a': 'Key 7df15425872b27bcf6e4b5bd6f2daee80cd1b42a not found in /da5_fast/All.sha1c/commit_125.tch'}


  6%|███▍                                                | 96/1477 [01:37<23:36,  1.03s/it]

Got Errors {'8dcdc2d4daba614aa67bad4d43f134c159d0818d': 'Key 8dcdc2d4daba614aa67bad4d43f134c159d0818d not found in /da5_fast/All.sha1c/commit_13.tch'}


  7%|███▌                                               | 102/1477 [01:44<23:22,  1.02s/it]

Got Errors {'9663b2004405032abf27a5ddb2b16f5bcbfd5f5b': 'Key 9663b2004405032abf27a5ddb2b16f5bcbfd5f5b not found in /da5_fast/All.sha1c/commit_22.tch'}


  9%|████▍                                              | 127/1477 [02:09<22:56,  1.02s/it]

Got Errors {'b956fd85001d5099959bf8de1805402f0fca1ed1': 'Key b956fd85001d5099959bf8de1805402f0fca1ed1 not found in /da5_fast/All.sha1c/commit_57.tch'}


  9%|████▌                                              | 132/1477 [02:14<22:50,  1.02s/it]

Got Errors {'c020c07169bf2e52d9be1559f1f8dba69d81216e': 'Key c020c07169bf2e52d9be1559f1f8dba69d81216e not found in /da5_fast/All.sha1c/commit_64.tch'}


  9%|████▋                                              | 136/1477 [02:18<22:46,  1.02s/it]

Got Errors {'c5416f05e2d5367d8af9af22cae4dcb6eeeb8104': 'Key c5416f05e2d5367d8af9af22cae4dcb6eeeb8104 not found in /da5_fast/All.sha1c/commit_69.tch'}


 10%|████▉                                              | 144/1477 [02:26<22:42,  1.02s/it]

Got Errors {'cf83cbe1ba077657d83015171fdf6baa550fc9c5': 'Key cf83cbe1ba077657d83015171fdf6baa550fc9c5 not found in /da5_fast/All.sha1c/commit_79.tch', 'cfdd2587fe0f1733647c7a4d4688c73eded97bcf': 'Key cfdd2587fe0f1733647c7a4d4688c73eded97bcf not found in /da5_fast/All.sha1c/commit_79.tch'}


 11%|█████▍                                             | 159/1477 [02:42<22:23,  1.02s/it]

Got Errors {'e548efd21f2c061f24635637222c76e60a74558d': 'Key e548efd21f2c061f24635637222c76e60a74558d not found in /da5_fast/All.sha1c/commit_101.tch'}


 11%|█████▌                                             | 162/1477 [02:45<22:20,  1.02s/it]

Got Errors {'ea45c2c4bf0b192d622eb45463ced87c663317ef': 'Key ea45c2c4bf0b192d622eb45463ced87c663317ef not found in /da5_fast/All.sha1c/commit_106.tch'}


 11%|█████▊                                             | 169/1477 [02:52<22:13,  1.02s/it]

Got Errors {'f5dcbd04a52de64cf39da831beb7fba41a41a182': 'Key f5dcbd04a52de64cf39da831beb7fba41a41a182 not found in /da5_fast/All.sha1c/commit_117.tch'}


 12%|█████▉                                             | 172/1477 [02:55<22:09,  1.02s/it]

Got Errors {'fb61c746ae9490c1611b57463994902aa40f42f1': 'Key fb61c746ae9490c1611b57463994902aa40f42f1 not found in /da5_fast/All.sha1c/commit_123.tch'}


 12%|█████▉                                             | 173/1477 [02:56<22:08,  1.02s/it]

Got Errors {'fd2800d9bc0e752fc9171c192c96152809ad4718': 'Key fd2800d9bc0e752fc9171c192c96152809ad4718 not found in /da5_fast/All.sha1c/commit_125.tch'}


 14%|███████▎                                           | 211/1477 [03:35<21:30,  1.02s/it]

Got Errors {'8f98a223de67a19e43414675fbcbcd17e0fc0b93': 'Key 8f98a223de67a19e43414675fbcbcd17e0fc0b93 not found in /da5_fast/All.sha1c/commit_15.tch'}


 22%|███████████                                        | 321/1477 [05:27<19:34,  1.02s/it]

Got Errors {'240252d845214e0208bdb2d6dd4f3086e8b82f37': 'Key 240252d845214e0208bdb2d6dd4f3086e8b82f37 not found in /da5_fast/All.sha1c/commit_36.tch'}


 38%|███████████████████▍                               | 564/1477 [09:34<15:30,  1.02s/it]

Got Errors {'060868c85b5468381b054bbc272e065f46ab7112': 'Key 060868c85b5468381b054bbc272e065f46ab7112 not found in /da5_fast/All.sha1c/commit_6.tch'}


 38%|███████████████████▌                               | 567/1477 [09:38<15:26,  1.02s/it]

Got Errors {'19496f7bfedb11333e14da45d4558128085d6e08': 'Key 19496f7bfedb11333e14da45d4558128085d6e08 not found in /da5_fast/All.sha1c/commit_25.tch'}


 38%|███████████████████▌                               | 568/1477 [09:39<15:26,  1.02s/it]

Got Errors {'209a07bafc588d15f6da7253247fd602e573b841': 'Key 209a07bafc588d15f6da7253247fd602e573b841 not found in /da5_fast/All.sha1c/commit_32.tch'}


 39%|███████████████████▋                               | 569/1477 [09:40<15:25,  1.02s/it]

Got Errors {'2922a25f643fb85ee046e08ae06fd6b857b31f57': 'Key 2922a25f643fb85ee046e08ae06fd6b857b31f57 not found in /da5_fast/All.sha1c/commit_41.tch'}


 39%|███████████████████▋                               | 571/1477 [09:42<15:23,  1.02s/it]

Got Errors {'2f33eab19638ddb40f0370bef8c05a4f638c080d': 'Key 2f33eab19638ddb40f0370bef8c05a4f638c080d not found in /da5_fast/All.sha1c/commit_47.tch', '2f642875b13942ebc99bb109e0ca2698908e437a': 'Key 2f642875b13942ebc99bb109e0ca2698908e437a not found in /da5_fast/All.sha1c/commit_47.tch'}


 39%|███████████████████▊                               | 574/1477 [09:45<15:20,  1.02s/it]

Got Errors {'3eca8c48dd09fd82035f95b417da4c843016abe0': 'Key 3eca8c48dd09fd82035f95b417da4c843016abe0 not found in /da5_fast/All.sha1c/commit_62.tch'}


 39%|████████████████████                               | 580/1477 [09:51<15:14,  1.02s/it]

Got Errors {'6428afc2dc6283964ee8426c5ed44137b3ee1745': 'Key 6428afc2dc6283964ee8426c5ed44137b3ee1745 not found in /da5_fast/All.sha1c/commit_100.tch'}


 39%|████████████████████                               | 581/1477 [09:52<15:13,  1.02s/it]

Got Errors {'6aa46d665c9f2c093bb7caab1914a475b64768c4': 'Key 6aa46d665c9f2c093bb7caab1914a475b64768c4 not found in /da5_fast/All.sha1c/commit_106.tch', '6c2c54a5a33554c886211373037810c30a2f8883': 'Key 6c2c54a5a33554c886211373037810c30a2f8883 not found in /da5_fast/All.sha1c/commit_108.tch'}


 39%|████████████████████                               | 582/1477 [09:53<15:12,  1.02s/it]

Got Errors {'71158dea57aeb37a6167167c73ff11c0f9b91f66': 'Key 71158dea57aeb37a6167167c73ff11c0f9b91f66 not found in /da5_fast/All.sha1c/commit_113.tch'}


 39%|████████████████████▏                              | 583/1477 [09:54<15:11,  1.02s/it]

Got Errors {'72fee439f74de904de5621d335a6d223853d841b': 'Key 72fee439f74de904de5621d335a6d223853d841b not found in /da5_fast/All.sha1c/commit_114.tch'}


 40%|████████████████████▏                              | 584/1477 [09:55<15:10,  1.02s/it]

Got Errors {'7552acc2ce223c86601e88ede727b382ffa00707': 'Key 7552acc2ce223c86601e88ede727b382ffa00707 not found in /da5_fast/All.sha1c/commit_117.tch', '7a4de2a20eae860dae05bb5b461acfb6228f781b': 'Key 7a4de2a20eae860dae05bb5b461acfb6228f781b not found in /da5_fast/All.sha1c/commit_122.tch'}


 40%|████████████████████▏                              | 585/1477 [09:56<15:09,  1.02s/it]

Got Errors {'7efa9f1e3412601f0d59c941146727e46824fc2f': 'Key 7efa9f1e3412601f0d59c941146727e46824fc2f not found in /da5_fast/All.sha1c/commit_126.tch', '8017aedafed762633243d024804826454d207bdf': 'Key 8017aedafed762633243d024804826454d207bdf not found in /da5_fast/All.sha1c/commit_0.tch'}


 40%|████████████████████▎                              | 587/1477 [09:58<15:07,  1.02s/it]

Got Errors {'8f3a6c7e6b1d479bd4b403ee2ec5056bcec71c1d': 'Key 8f3a6c7e6b1d479bd4b403ee2ec5056bcec71c1d not found in /da5_fast/All.sha1c/commit_15.tch'}


 40%|████████████████████▎                              | 588/1477 [09:59<15:06,  1.02s/it]

Got Errors {'9656fa751e4f5f879234c105c82d3718e7b0036d': 'Key 9656fa751e4f5f879234c105c82d3718e7b0036d not found in /da5_fast/All.sha1c/commit_22.tch'}


 40%|████████████████████▍                              | 593/1477 [10:04<15:00,  1.02s/it]

Got Errors {'af74cc29cb9fe31f47fcd63eca98d67c94d6c499': 'Key af74cc29cb9fe31f47fcd63eca98d67c94d6c499 not found in /da5_fast/All.sha1c/commit_47.tch'}


 40%|████████████████████▌                              | 594/1477 [10:05<14:59,  1.02s/it]

Got Errors {'b6a53c279d97a5a289273401ef7c816cc65054a6': 'Key b6a53c279d97a5a289273401ef7c816cc65054a6 not found in /da5_fast/All.sha1c/commit_54.tch'}


 40%|████████████████████▌                              | 595/1477 [10:06<15:00,  1.02s/it]

Got Errors {'bb6033867f844f7c873d8bce9ea1cd0edc261766': 'Key bb6033867f844f7c873d8bce9ea1cd0edc261766 not found in /da5_fast/All.sha1c/commit_59.tch'}


 40%|████████████████████▋                              | 598/1477 [10:09<14:57,  1.02s/it]

Got Errors {'cd9b79c12128a1571fba1b89a8cd0def9e715ed4': 'Key cd9b79c12128a1571fba1b89a8cd0def9e715ed4 not found in /da5_fast/All.sha1c/commit_77.tch'}


 41%|████████████████████▋                              | 600/1477 [10:11<14:54,  1.02s/it]

Got Errors {'d607ed6660f6979efb52843b2cae77000240a616': 'Key d607ed6660f6979efb52843b2cae77000240a616 not found in /da5_fast/All.sha1c/commit_86.tch', 'd6ab7950a8d75b1033036c347dbb4215f682d909': 'Key d6ab7950a8d75b1033036c347dbb4215f682d909 not found in /da5_fast/All.sha1c/commit_86.tch'}


 41%|████████████████████▊                              | 601/1477 [10:12<14:53,  1.02s/it]

Got Errors {'dd2125e313e3c9f56719a264e94d1ba9fbdd9ba4': 'Key dd2125e313e3c9f56719a264e94d1ba9fbdd9ba4 not found in /da5_fast/All.sha1c/commit_93.tch'}


 41%|████████████████████▊                              | 602/1477 [10:13<14:52,  1.02s/it]

Got Errors {'e02f616faf1b0bff9c6a8bbf3fdf5915cf7fca32': 'Key e02f616faf1b0bff9c6a8bbf3fdf5915cf7fca32 not found in /da5_fast/All.sha1c/commit_96.tch'}


 41%|████████████████████▊                              | 604/1477 [10:15<14:50,  1.02s/it]

Got Errors {'ec9e318b88e0255dcf349a0493b053307bcc49c9': 'Key ec9e318b88e0255dcf349a0493b053307bcc49c9 not found in /da5_fast/All.sha1c/commit_108.tch'}


 41%|████████████████████▉                              | 605/1477 [10:16<14:49,  1.02s/it]

Got Errors {'f50a6a8cfa4cff827028b912e256621eda1fa2cf': 'Key f50a6a8cfa4cff827028b912e256621eda1fa2cf not found in /da5_fast/All.sha1c/commit_117.tch'}


 41%|████████████████████▉                              | 606/1477 [10:17<14:48,  1.02s/it]

Got Errors {'fcdf42c4832412aac375b7cbf80448b79dfdf7b0': 'Key fcdf42c4832412aac375b7cbf80448b79dfdf7b0 not found in /da5_fast/All.sha1c/commit_124.tch', 'fd80a74ffb454fafb1f05ecb329ffaf3e9a1639a': 'Key fd80a74ffb454fafb1f05ecb329ffaf3e9a1639a not found in /da5_fast/All.sha1c/commit_125.tch'}


 81%|████████████████████████████████████████▍         | 1194/1477 [20:28<04:54,  1.04s/it]

Got Errors {'0e93fb614dfe88868a196daf655cc1929e51f9e9': 'Key 0e93fb614dfe88868a196daf655cc1929e51f9e9 not found in /da5_fast/All.sha1c/commit_14.tch'}


 84%|█████████████████████████████████████████▊        | 1236/1477 [21:12<04:12,  1.05s/it]

Got Errors {'39ef56f85d605ec029a273e3a5228bfbffeb3d82': 'Key 39ef56f85d605ec029a273e3a5228bfbffeb3d82 not found in /da5_fast/All.sha1c/commit_57.tch'}


 84%|██████████████████████████████████████████▏       | 1246/1477 [21:22<04:03,  1.06s/it]

Got Errors {'43dfeb5fe4b89676c14fbe1845ef9611a0dae3a6': 'Key 43dfeb5fe4b89676c14fbe1845ef9611a0dae3a6 not found in /da5_fast/All.sha1c/commit_67.tch', '443a24526a8d2cbdbd78cb3ad059e9123489604b': 'Key 443a24526a8d2cbdbd78cb3ad059e9123489604b not found in /da5_fast/All.sha1c/commit_68.tch'}


 85%|██████████████████████████████████████████▎       | 1249/1477 [21:25<04:00,  1.05s/it]

Got Errors {'46c1f7069b029157bf15b28778fd937053595790': 'Key 46c1f7069b029157bf15b28778fd937053595790 not found in /da5_fast/All.sha1c/commit_70.tch'}


 87%|███████████████████████████████████████████▎      | 1279/1477 [21:57<03:27,  1.05s/it]

Got Errors {'671080328e287ddafe46723920ef1b150d0d11f0': 'Key 671080328e287ddafe46723920ef1b150d0d11f0 not found in /da5_fast/All.sha1c/commit_103.tch'}


 90%|█████████████████████████████████████████████▏    | 1333/1477 [22:54<02:30,  1.05s/it]

Got Errors {'9d73557ca808b50ae845bbc8de19fecd73fdc2bd': 'Key 9d73557ca808b50ae845bbc8de19fecd73fdc2bd not found in /da5_fast/All.sha1c/commit_29.tch'}


 91%|█████████████████████████████████████████████▋    | 1349/1477 [23:10<02:12,  1.04s/it]

Got Errors {'ad0e36924de0a2c337888c6279a6154d73b909bb': 'Key ad0e36924de0a2c337888c6279a6154d73b909bb not found in /da5_fast/All.sha1c/commit_45.tch'}


 92%|█████████████████████████████████████████████▊    | 1355/1477 [23:17<02:06,  1.04s/it]

Got Errors {'b40edd3d63b8df7bf68c7bcf623ee1fd63b98440': 'Key b40edd3d63b8df7bf68c7bcf623ee1fd63b98440 not found in /da5_fast/All.sha1c/commit_52.tch'}


 93%|██████████████████████████████████████████████▎   | 1369/1477 [23:31<01:51,  1.04s/it]

Got Errors {'c28391e1faa3aa22400ec44e6c41745ea4780bb3': 'Key c28391e1faa3aa22400ec44e6c41745ea4780bb3 not found in /da5_fast/All.sha1c/commit_66.tch'}


 94%|███████████████████████████████████████████████   | 1392/1477 [23:55<01:28,  1.04s/it]

Got Errors {'da5ed37cc07439327b4961421e9b4b4582fcdd00': 'Key da5ed37cc07439327b4961421e9b4b4582fcdd00 not found in /da5_fast/All.sha1c/commit_90.tch'}


 96%|███████████████████████████████████████████████▉  | 1415/1477 [24:19<01:04,  1.03s/it]

Got Errors {'f155caaabfb29d8fc8ae837e1c5814e1436bfff8': 'Key f155caaabfb29d8fc8ae837e1c5814e1436bfff8 not found in /da5_fast/All.sha1c/commit_113.tch'}


 96%|███████████████████████████████████████████████▉  | 1417/1477 [24:21<01:02,  1.04s/it]

Got Errors {'f2aa5c5bb2a504e4b2a457fa688bd35f57f536b0': 'Key f2aa5c5bb2a504e4b2a457fa688bd35f57f536b0 not found in /da5_fast/All.sha1c/commit_114.tch'}


 96%|████████████████████████████████████████████████  | 1418/1477 [24:22<01:01,  1.04s/it]

Got Errors {'f3e867154d91c63cf82557b2b4c354384f0aa367': 'Key f3e867154d91c63cf82557b2b4c354384f0aa367 not found in /da5_fast/All.sha1c/commit_115.tch'}


100%|██████████████████████████████████████████████████| 1477/1477 [25:24<00:00,  1.03s/it]


,commit,tree,parent,author,author_time,author_tz,committer,committer_time,committer_tz,message,sha1,project
0,000287679e1bef118cffcc087810c3fea90b6dfe,a84e7e257a5846ce8ab8f25a4552e969f54682c4,[80fcf7f7ba373373cd8f07c923bf71adae751bf7],Peter Williams <peter@newton.cx>,1602898244,-0400,Peter Williams <peter@newton.cx>,1602898244,-0400,ci/azure-build-and-test.yml: skip Windows/Pyth...,000287679e1bef118cffcc087810c3fea90b6dfe,worldwidetelescope_pywwt
1,00160f35b811acb815ef8f83be5b8b08450699cc,7d95b8443f8388457efed560f10f9fb51f2c6352,[3ef4ec9a3dd615ebf4808cb8a3a468dc34a679e1],Peter Williams <peter@newton.cx>,1735501022,-0500,Peter Williams <peter@newton.cx>,1735502749,-0500,ci/azure-build-and-test.yml: update to macos-1...,00160f35b811acb815ef8f83be5b8b08450699cc,worldwidetelescope_pywwt


In [37]:
# fixed to use df_commit_data to fix project assignemnt error

# what does the first record looks like?
# commit  parents
v = df_commit_data.iloc[0]
# commit tree, [parent{s}], [author, unixtime, tz ], [ committer, unixtime, tz ], Commit_message ]
print(v)

commit                     000287679e1bef118cffcc087810c3fea90b6dfe
tree                       a84e7e257a5846ce8ab8f25a4552e969f54682c4
parent                   [80fcf7f7ba373373cd8f07c923bf71adae751bf7]
author                             Peter Williams <peter@newton.cx>
author_time                                              1602898244
author_tz                                                     -0400
committer                          Peter Williams <peter@newton.cx>
committer_time                                           1602898244
committer_tz                                                  -0400
message           ci/azure-build-and-test.yml: skip Windows/Pyth...
sha1                       000287679e1bef118cffcc087810c3fea90b6dfe
project                                    worldwidetelescope_pywwt
Name: 0, dtype: object


In [38]:
# fixed to use df_commit_data to fix project assignemnt error

# Now that the data is retrieved, save it and commit to your GH fork
# First, we need to flatten info in order to export as csv
# our dataframe will have columns project,'commit, author, time, message'
dfinf = pd.DataFrame(columns=['project', 'commit', 'author', 'time', 'message'])
for i, k in df_commit_data.iterrows():
  row = pd.Series({'project':k['project'], 'commit': k['commit'], 'author': k['author'], 'time': k['author_time'], 'message':k['message']})
  dfinf = pd.concat([dfinf, row.to_frame().T ], ignore_index=True)


In [39]:
# check if it has the right content
dfinf.head(1)

,project,commit,author,time,message
0,worldwidetelescope_pywwt,000287679e1bef118cffcc087810c3fea90b6dfe,Peter Williams <peter@newton.cx>,1602898244,ci/azure-build-and-test.yml: skip Windows/Pyth...


In [36]:
#mode a means append, so you have all your projects in the same file
yournetid='hneel'
dfinf.to_csv(yournetid+'_project_summary.csv', index=False,sep=';', mode='a', header=False)

# Make sure you check in to your fork not just the notebook but also the csv files!!!

# Don't forget to add requested data from github and this notebook
### For each of the 10 projects go to their github repo and get the number of stars, number of forks, and the last commit date
### Report (in your notebook) the number of commits, the number of authors, and max and min time for each project based on WoC commits and also add the info you obtained from github

#### WorldWideTelescope/pywwt
- Stars: 55
- Forks: 14
- Last Commit Date: July 11th, 2026

#### chakravala/Grassmann.jl
- Stars: 515
- Forks: 41
- Last Commit Date: August 9th, 2026

#### roboticslibrary/rl
- Stars: 1.2k
- Forks: 244
- Last Commit Date: April 15th, 2025

#### ROBelgium/MSNoise
- Stars: 214
- Forks: 94
- Last Commit Date: August 20th, 2026

#### stepjam/RLBench
- Stars: 1.8k
- Forks: 323
- Last Commit Date: January 25th, 2025

#### FragIt/fragit-main
- Stars: 28
- Forks: 15
- Last Commit Date: April 11th, 2025

#### architecture-building-systems/revitpythonshell
- Stars: 595
- Forks: 125
- Last Commit Date: April 11th, 2025

#### VLSIDA/OpenRAM
- Stars: 1.1k
- Forks: 288
- Last Commit Date: August 16th, 2026

#### kaczmarj/neurodocker
- Stars: 361
- Forks: 99
- Last Commit Date: September 23rd, 2026 

#### code-iai/ROSIntegration
- Stars: 468
- Forks: 142
- Last Commit Date: November 3rd, 2025

In [56]:
for prj in projects:
  print(prj)
  prj = prj.lower().replace('/','_',2)
  mask = dfinf['project'] == prj
  temp = dfinf[mask]
  print("Commits:", len(temp))
  print("Authors:", temp['author'].nunique())
  print("Max Time:", temp['time'].max())
  print("Min Time:", temp['time'].min())
  print("")

WorldWideTelescope/pywwt
Commits: 1725
Authors: 16
Max Time: 1757020950
Min Time: 1389625907

chakravala/Grassmann.jl
Commits: 597
Authors: 15
Max Time: 1760912173
Min Time: 1503791400

roboticslibrary/rl
Commits: 578
Authors: 20
Max Time: 1750643853
Min Time: 1402513214

ROBelgium/MSNoise
Commits: 2084
Authors: 52
Max Time: 1776211363
Min Time: 1377438755

stepjam/RLBench
Commits: 619
Authors: 38
Max Time: 1756280121
Min Time: 1569533042

FragIt/fragit-main
Commits: 406
Authors: 5
Max Time: 1747942920
Min Time: 1335610324

architecture-building-systems/revitpythonshell
Commits: 656
Authors: 29
Max Time: 1744359920
Min Time: 1258533985

VLSIDA/OpenRAM
Commits: 5074
Authors: 129
Max Time: 1761069676
Min Time: 1478107108

kaczmarj/neurodocker
Commits: 2490
Authors: 87
Max Time: 1761955608
Min Time: 1487367191

code-iai/ROSIntegration
Commits: 467
Authors: 48
Max Time: 1762166372
Min Time: 1501619346

